tests/5.1/default/test_default_firstprivate_task.c

```diff
@@ -0,0 +1,58 @@
+//--------------- test_default_firstprivate_task.c -------------------------//
+//
+// OpenMP API Version 5.1 Aug 2021
+//
+// Tests the behavior of the default clause when the specified data-sharing attribute is firstprivate. This test focuses 
+// on the use of the task construct. Firstprivate should allow any data initialized outside the task region to return to its original value
+// after being manipulated within the task region.
+
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+#include <math.h>
+
+#define N 1024
+
+int errors, i;
+
+int test_default_firstprivate_task(){
+	int test_num = 1;
+	int test_arr[N];
+	int sum = 0;
+	for (int i = 0; i<N; i++){
+		test_arr[i] = i;
+		sum += i;
+	}
+
+	#pragma omp task default(firstprivate)
+	{
+		test_num += 1;
+		for (int i=0; i<N; i++){
+			test_arr[i] = 1;
+		}
+	}
+	int new_sum;
+	int wrong_sum;
+	for (int i = 0; i<N; i++){
+		new_sum += test_arr[i];
+		wrong_sum += 1;
+	}
+	printf("%d",test_num);
```

Comment:
We usually try to take any print statements out of the test before we merge them! If they are useful for discussion and explanation of the test that is o.k. but ultimately try to use OMPVV_INFOMSG instead of printf (calls printf under the hood anywho)